### ISSUE TITLE
Model parameters become NaN during training

### ISSUE THREAD

**Issue by JamChaos** [OpenRobotLab/AnySplat#99]
I am encountering an issue where model parameters suddenly turn into NaN during training, leading to a complete crash.
<img width="737" height="33" alt="Image" src="https://github.com/user-attachments/assets/7c3d54c3-b749-4360-b883-7408ea2fb25e" />
I have noticed that the recorded grad explode significantly.

<img width="720" height="336" alt="Image" src="https://github.com/user-attachments/assets/da6cadc5-19f0-4e79-a5f3-85103eacf66b" />

What makes this strange is that I have already implemented the following safeguards, yet the issue persists:
1. Gradient Clipping is active.
2. Loss Handling: Logic is in place to zero out NaN losses.
3. Loss Clipping: Loss values are capped within a specific range.

Despite these, the parameters still become corrupted. I noticed other users have reported similar issues in this repository, suggesting this might be a systemic numerical instability within the architecture or the specific layer implementation or specific scene.

**Comment by myc634** [comment]
We encountered the same issue during training, simply skip iter which the loss value of NaN will help
